### PR TITLE
Clarify TestRateLimit

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR cleans up some of the bits of how the rate limit is evaluated and tested.

**Release note**:

```release-note-none
NONE
```

### CODE REVIEW

**Comment by douglas-reid** [comment]
ptal. i switched `TestRateLimit` to use the v3-only rule (instead of trying to split between v2/v3) which should make the value inspection simpler. i also added promDump logging to all appropriate tests to make it easier to pinpoint failures better (on the mixer side).

File: tests/e2e/tests/mixer/mixer_test.go [hunk 1]
```diff
@@ -163,16 +163,17 @@ func TestGlobalCheckAndReport(t *testing.T) {
 	if err := visitProductPage(testRetryTimes); err != nil {
 		t.Fatalf("Test app setup failure: %v", err)
 	}
+	// must sleep to allow for prometheus scraping, etc.
+	time.Sleep(30 * time.Second)
```
Comment (ldemailly) [inline tests/e2e/tests/mixer/mixer_test.go]: nit: maybe log before sleep is better to know the pause is normal

File: tests/e2e/tests/mixer/mixer_test.go [hunk 2]
```diff
@@ -202,21 +205,22 @@ func TestNewMetrics(t *testing.T) {
 
 	// allow time for configuration to go and be active.
 	// TODO: figure out a better way to confirm rule active
-	time.Sleep(5 * time.Second)
+	time.Sleep(30 * time.Second)
```
Comment (ldemailly) [inline tests/e2e/tests/mixer/mixer_test.go]: nit: that 30s could be a constant sleepTime or such

File: tests/e2e/tests/mixer/mixer_test.go [hunk 3]
```diff
@@ -95,7 +95,7 @@ func createDefaultRoutingRules() error {
 	if err := createRouteRule(routeAllRule); err != nil {
 		return fmt.Errorf("could not create base routing rules: %v", err)
 	}
-	time.Sleep(30 * time.Second)
+	allowRuleSync()
```
Comment (ldemailly) [inline tests/e2e/tests/mixer/mixer_test.go]: 👍

File: tests/e2e/tests/mixer/mixer_test.go [hunk 4]
```diff
@@ -344,33 +366,44 @@ func TestRateLimit(t *testing.T) {
...
 	}
 }
 
+func allowRuleSync() {
+	glog.Info("Sleeping to allow rules to take effect...")
+	time.Sleep(1 * time.Minute)
+}
+
+func allowPrometheusSync() {
+	glog.Info("Sleeping to allow prometheus to record metrics...")
```
Comment (ldemailly) [inline tests/e2e/tests/mixer/mixer_test.go]: :+1:
(we could also have our own sleep that says how long it's about to sleep with optional "why" message)

**Comment by douglas-reid** [comment]
I ran this twice in a row on my private cluster. Both times:

```
****************************************************
Tests Summary
PASSED: //tests/e2e/tests/mixer:go_default_test
PASSED: //tests/e2e/tests/bookinfo:go_default_test
```

**Comment by douglas-reid** [comment]
I'll wait to see what `prow/e2e-suite-no_rbac-no_auth.sh` does, but either way, i'm merging unless there is an objection.

**Pull request merged**